### PR TITLE
refactor settleAction() in order to simplify CandidateStateManager in…

### PR DESCRIPTION
**Pull request by lzxm160**
…terface 2087

work on #2087

### CODE REVIEW

File: action/protocol/managers.go [hunk 1]
```diff
@@ -76,16 +76,26 @@ type (
...
 		ReadView(string) (interface{}, error)
 	}
+	
+	// StateWriter defines an interface to write stateDB
+	StateWriter interface {
+		PutState(interface{}, ...StateOption) (uint64, error)
+		DelState(...StateOption) (uint64, error)
+	}
 
+	// StateReadWriter defines an interface to read and write stateDB
```
Comment (CoderZhi) [inline action/protocol/managers.go:86]: bad naming

File: action/protocol/managers.go [hunk 2]
```diff
@@ -76,16 +76,26 @@ type (
 		States(...StateOption) (uint64, state.Iterator, error)
 		ReadView(string) (interface{}, error)
 	}
+	
+	// StateWriter defines an interface to write stateDB
+	StateWriter interface {
```
Comment (CoderZhi) [inline action/protocol/managers.go:81]: it is better to have
```
StateWriter interface {
    StateReader
    PutState...
    DelState...
}
```

but I don't see the value distinguishing the statewriter and statemanager.

File: action/protocol/staking/candidate_statemanager.go [hunk 3]
```diff
@@ -30,6 +29,9 @@ type (
 		GetBySelfStakingIndex(uint64) *Candidate
 		Upsert(*Candidate) error
 		Commit() error
+		// bucket-related
+		protocol.StateReadWriter
+		Load(string, interface{}) error
```
Comment (CoderZhi) [inline action/protocol/staking/candidate_statemanager.go:34]: if this is the purpose you create new interface, I think you are not on the right track. Our goal is to remove protocol.StateXXX from this interface definition.

Comment (lzxm160) [reply action/protocol/staking/candidate_statemanager.go:34]: There's some interfaces cannot be removed in protocol.StateReadWriter,like State,PutState,DelState,used in handleXXX

File: action/protocol/managers.go (hunk 2 above)
Comment (lzxm160) [reply action/protocol/managers.go:81]: sure,I can change it to this

**Comment by lzxm160** [comment]
so many state manager's interface to add,like putstate,delstate,and depositGas used protocol.StateManager,LoadAccount used protocol.StateReader

**Pull request closed without merging**